src/explorer/SiteTreeItem.ts

```diff
@@ -161,7 +161,7 @@ export abstract class SiteTreeItem implements IAzureParentTreeItem {
 
     private createLabel(state: string): string {
         return (this.client.slotName ? this.client.slotName : this.client.siteName) +    // Site/slot name
-            (state && state.toLowerCase() !== 'running' ? ` (${state})` : '');  // Status (if site/slot not running)
+            (state && state !== undefined && state.toLowerCase() !== 'running' ? ` (${state})` : '');  // Status (if site/slot not running)
```

Comment:
`undefined` is a false-y value, so `state && state !== undefined ...` should be equivalent to just `state && ...`.

Do you have repro steps for this bug? I'm wondering if there's something else that needs to be fixed

Reply:
I do, yes. When connecting to a subscription where I don't have full permissions on all the web app resources, the "state" will not be available for some items. When this occurs, I only get "Error: Cannot read property 'toLowerCase' of undefined" in the explorer window:

VS-Code explorer window:
![image](https://user-images.githubusercontent.com/2287216/37994586-0142b2fe-31e0-11e8-9fc8-769d3e94d28f.png)

Web Apps in the Portal (note how I can see the Status on some, but not all):
![image](https://user-images.githubusercontent.com/2287216/37994673-43362060-31e0-11e8-8776-1af401ed7b80.png)

Reply:
Do you mind sharing what role you have on those resources? We tried 'reader' and 'wesite contributor' but I can still see the status

Here's what I see for the options:
![screen shot 2018-03-27 at 3 39 59 pm](https://user-images.githubusercontent.com/11282622/37998885-55fc3d16-31d5-11e8-8d32-f07ec0712994.png)

It should be listed if you check 'Access Control' under the resource

Reply:
So, it's a custom role, not one of the built in ones, which, interestingly enough, is the same role for App Services that I can see the status for...

Reply:
A little more info... if I click on all the App Services in the list, there are 3 that appear like the below... think that may be the issue?

![image](https://user-images.githubusercontent.com/2287216/37999543-5650e5fc-31f1-11e8-81e7-65c4d92218ee.png)

![image](https://user-images.githubusercontent.com/2287216/37999563-739ba8cc-31f1-11e8-9b34-8cf450c20979.png)

Reply:
That certainly could be a part of the problem. I'm not sure how familiar you are with VS Code extension development - but can you F5 this repo and find out where the exception is being thrown? Again - I'm skeptical that it's at this location, but there could easily be another 'toLowerCase' either in this repo or in one of our dependencies

I would turn on 'All Exceptions' in the debug view _right_ before you expand the subscription node in the tree. If you do it too soon, you'll see a bunch of VS Code exceptions that aren't relevant
![screen shot 2018-03-27 at 4 20 12 pm](https://user-images.githubusercontent.com/11282622/38000089-c63a1ab2-31da-11e8-9209-2c62601822da.png)

Reply:
Ok... running from the repo changed the error message a bit ("Cannot read property 'match' of undefined')... (sorry for all the screenshots... not sure what the best way to get this to you is)

![image](https://user-images.githubusercontent.com/2287216/38000649-9916e6ac-31f6-11e8-9b68-912440822f55.png)

![image](https://user-images.githubusercontent.com/2287216/38000664-a91eafda-31f6-11e8-94a2-36b94c931db7.png)

![image](https://user-images.githubusercontent.com/2287216/38000676-bcc7eccc-31f6-11e8-8c33-170e08d91d83.png)

![image](https://user-images.githubusercontent.com/2287216/38000688-cae42f8c-31f6-11e8-8c06-695a7080dad8.png)

![image](https://user-images.githubusercontent.com/2287216/38000699-d903863a-31f6-11e8-8172-b80597e7433d.png)

![image](https://user-images.githubusercontent.com/2287216/38000721-f204be88-31f6-11e8-86d8-05f08af0fde1.png)

![image](https://user-images.githubusercontent.com/2287216/38000735-fff78b06-31f6-11e8-98c9-961f38901d1e.png)

![image](https://user-images.githubusercontent.com/2287216/38000748-0f756f30-31f7-11e8-8c07-30bf4bae8d37.png)

![image](https://user-images.githubusercontent.com/2287216/38000764-21c5821a-31f7-11e8-86e7-4f442ba886d1.png)

Reply:
So, apparently, the serverFarmId for some of the resources in our subscription is blank. Not sure how that happened or if we need to clean those resources up or not. That being said, I'll play around with the code here to see if there is any way I can get it to return something valid (maybe skip those invalid resources?) when this occurs, so that I can still use the extension.

If it's ok with you, I'll close this PR and start a new one after I try to find a better solution.

Reply:
Yeah I think you're right. We've changed the code since 0.6.1, but in both cases its the serverFarmId that's undefined. Without a serverFarmId, you'll definitely have limited functionality for those resources. That being said, we should still display the resources that you _do_ have a serverFarmId for. 

Please file an issue on this repo to track the work. If you want to tackle it, add a note in the issue so that we know. Otherwise we can take a look